src/couch_views/src/couch_views.erl

```diff
@@ -53,11 +54,15 @@ query(Db, DDoc, ViewName, Callback, Acc0, Args0) ->
 
     try
         fabric2_fdb:transactional(Db, fun(TxDb) ->
-            ok = maybe_update_view(TxDb, Mrst, Args3),
+            if IsInteractive == true -> ok; true ->
+                ok = maybe_update_view(TxDb, Mrst, Args3)
+            end,
             read_view(TxDb, Mrst, ViewName, Callback, Acc0, Args3)
         end)
     catch throw:{build_view, WaitSeq} ->
-        couch_views_jobs:build_view(Db, Mrst, WaitSeq),
+        % make sure the job is not added in the same transaction
```

Comment:
And it just occurred to me that this is broken. I can't remember what it was that caused this in the first place, but by doing this out of transaction we're left with the ability to create jobs for design docs that may have failed to be committed for whatever reason.

Reply:
I've fixed this now. I've moved the transaction logic for build_views into `build_view_async`